case-studies/higgs/mH-recoil/FCCAnalyses-config/APC/analysis.py

```diff
@@ -0,0 +1,146 @@
+import sys
+import ROOT
+
+print ("Load cxx analyzers ... ",)
+ROOT.gSystem.Load("libedm4hep")
+ROOT.gSystem.Load("libpodio")
+ROOT.gSystem.Load("libFCCAnalyses")
+ROOT.gSystem.Load("libFCCAnalysesHiggs")
+ROOT.gErrorIgnoreLevel = ROOT.kFatal
+_edm  = ROOT.edm4hep.ReconstructedParticleData()
+_pod  = ROOT.podio.ObjectID()
+_fcc  = ROOT.dummyLoader
+_higgs  = ROOT.dummyLoaderHiggs
+print ('edm4hep  ',_edm)
+print ('podio    ',_pod)
+print ('fccana   ',_fcc)
+print ('higgs   ',_higgs)
+
+class analysis():
+
+    #__________________________________________________________
+    def __init__(self, inputlist, outname, ncpu):
+        self.outname = outname
+        if ".root" not in outname:
+            self.outname+=".root"
+
+        ROOT.ROOT.EnableImplicitMT(ncpu)
+
+        self.df = ROOT.RDataFrame("events", inputlist)
+        print (" done")
+    #__________________________________________________________
+    def run(self):
+        df2 = (self.df
+               # define an alias for muon index collection
+               .Alias("Muon0", "Muon#0.index")
+               # define the muon collection
+               .Define("muons",  "ReconstructedParticle::get(Muon0, ReconstructedParticles)")
+               # define missing momentum
+               .Define('missingET_px', 'MissingET.momentum.x')
+               .Define('missingET_py', 'MissingET.momentum.y')
+               .Define('missingET_pz', 'MissingET.momentum.z')
+               .Define('missingET_e', 'MissingET.energy')
+               # get cosTheta miss
+               .Define('missingET_costheta', 'APCHiggsTools::get_cosTheta_miss(missingET_px,missingET_py,missingET_pz,missingET_e)')
```

Comment:
You could simply do:
.Define('missing_cosTheta',  'APCHiggsTools::get_cosTheta(  MissingET )' )